packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.component.tsx

```diff
@@ -0,0 +1,198 @@
+
+  const onBeforeGetContentResolve = useRef(null);
+
+  useEffect(() => {
+    if (isPrinting && onBeforeGetContentResolve.current) {
+      onBeforeGetContentResolve.current();
+    }
+  }, [isPrinting]);
+
+  const patientDetails = useMemo(() => {
+    const getGender = (gender: string): string => {
+      switch (gender) {
+        case 'male':
+          return t('male', 'Male');
+        case 'female':
+          return t('female', 'Female');
+        case 'other':
+          return t('other', 'Other');
+        case 'unknown':
+          return t('unknown', 'Unknown');
+        default:
+          return gender;
+      }
+    };
+
+    const identifiers =
+      patient?.identifier?.filter(
+        (identifier) => !excludePatientIdentifierCodeTypes?.uuids.includes(identifier.type.coding[0].code),
+      ) ?? [];
+
+    return {
+      id: patient?.id,
+      photo: patient?.photo,
+      name: patient ? displayName(patient) : '',
+      dateOfBirth: patient.birthDate,
+      age: age(patient?.birthDate),
+      gender: getGender(patient?.gender),
+      address: patient?.address,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => ({ value, type })) : [],
+    };
+  }, [patient, t, excludePatientIdentifierCodeTypes?.uuids]);
+  const handleBeforeGetContent = useCallback(() => {
+    return new Promise<void>((resolve) => {
+      if (patient && headerTitle) {
+        onBeforeGetContentResolve.current = resolve;
+        setIsPrinting(true);
+        const printStyles = `@media print { @page { size: ${printIdentifierStickerSize}; } }`;
+
+        const style = document.createElement('style');
+        style.appendChild(document.createTextNode(printStyles));
+
+        document.head.appendChild(style);
+      }
+    });
+  }, [patient, printIdentifierStickerSize, headerTitle]);
+
+  const handleAfterPrint = useCallback(() => {
+    onBeforeGetContentResolve.current = null;
+    setIsPrinting(false);
+    closeModal();
+  }, [closeModal]);
+
+  const handlePrint = useReactToPrint({
+    content: () => contentToPrintRef.current,
+    documentTitle: `${patientDetails.name} - ${headerTitle}`,
+    onBeforeGetContent: handleBeforeGetContent,
+    onAfterPrint: handleAfterPrint,
+    onPrintError: (errorLocation, error) => {
+      new Promise<void>((resolve) => {
+        if (error) {
+          if (printRetryCount < 2) {
+            setprintRetryCount(printRetryCount + 1);
+            handleBeforeGetContent().then(() => {
+              handlePrint();
+            });
+          } else {
+            showSnackbar({
+              title: t('printFailed', 'Print Failed'),
+              subtitle: `${t('printingFailed', 'Printing has failed after 3 attempts')}`,
+              kind: 'error',
+              isLowContrast: true,
+            });
+            setprintRetryCount(1);
+          }
+          resolve();
+        }
+      });
+    },
+  });
+
+  const renderElementsInPairs = (elements) => {
```

Comment:
We should convert this from a function to a component. That will behave better with React/